xde/src/xde.rs

```diff
@@ -2392,15 +2393,13 @@ fn handle_mcast_tx<'a>(
                 ipv6_bytes[15],
             ]);
 
-            // Fill in outer MAC addresses
-            let final_pkt = unsafe {
-                let mblk = fwd_pkt.unwrap_mblk().as_ptr();
-                let rptr = (*mblk).b_rptr;
-                ptr::copy(dst_mac.as_ptr(), rptr, 6);
-                ptr::copy(mac_src.as_ptr(), rptr.add(6), 6);
-
-                MsgBlk::wrap_mblk(mblk).unwrap()
-            };
+            // Fill in outer MAC addresses.
+            // In this case we know we have pulled up the outer headers,
+            // so we are guaranteed enough space to have Ethernet source/destination
+            // contiguous.
+            fwd_pkt[..ETHER_ADDR_LEN].copy_from_slice(&*dst_mac);
+            fwd_pkt[ETHER_ADDR_LEN..2 * ETHER_ADDR_LEN]
```

Comment:
nit: the pattern you use below would be nice here, same at 2966

```rust
fwd_pkt[ETHER_ADDR_LEN..][..ETHER_ADDR_LEN].copy_from_slice(&*src_mac);
```

Reply:
Good call, thanks for the quick review.